pool/pool.go

```diff
@@ -50,35 +50,38 @@ type Client interface {
 	SessionCreate(context.Context, client.PrmSessionCreate) (*client.ResSessionCreate, error)
 }
 
-// BuilderOptions contains options used to build connection Pool.
-type BuilderOptions struct {
+// InitParameters contains options used to create connection Pool.
+type InitParameters struct {
 	Key                       *ecdsa.PrivateKey
 	Logger                    *zap.Logger
 	NodeConnectionTimeout     time.Duration
 	NodeRequestTimeout        time.Duration
 	ClientRebalanceInterval   time.Duration
 	SessionTokenThreshold     time.Duration
 	SessionExpirationDuration uint64
-	nodesParams               []*NodesParam
-	clientBuilder             func(endpoint string) (Client, error)
+	NodeParams                []NodeParam
+
+	clientBuilder func(endpoint string) (Client, error)
+}
+
+type rebalanceParameters struct {
+	nodesParams               []*nodesParam
+	nodeRequestTimeout        time.Duration
+	clientRebalanceInterval   time.Duration
+	sessionExpirationDuration uint64
 }
 
-type NodesParam struct {
+type nodesParam struct {
 	priority  int
 	addresses []string
 	weights   []float64
 }
 
+// NodeParam groups parameters of remote node.
 type NodeParam struct {
-	priority int
-	address  string
-	weight   float64
-}
-
-// Builder is an interim structure used to collect node addresses/weights and
-// build connection Pool subsequently.
-type Builder struct {
-	nodeParams []NodeParam
+	Priority int
```

Comment:
It's hard to support compatibility of exported fields, so I suggest to provide setters. Otherwise, all fields should be documented.